### PR TITLE
Add flexible postgres for et preview

### PR DESCRIPTION
**Pull request by dharmendrak**

### CODE REVIEW

File: apps/et/preview/aso/et-postgres.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+metadata:
+  name: ${NAMESPACE}-flexibleserver-postgres-${ENVIRONMENT}
+  namespace: ${NAMESPACE}
+  annotations:
+    serviceoperator.azure.com/reconcile-policy: detach-on-delete
+spec:
+  owner:
+    name: ${NAMESPACE}-aso-${ENVIRONMENT}-rg  #created ahead and added to flux
+  location: uksouth
+  version: "11"
```
Comment (hsjhita) [inline apps/et/preview/aso/et-postgres.yaml:12]: would it be better to configure this to v14?

File: apps/et/preview/sops-secrets/kustomization.yaml [hunk 2]
```diff
@@ -2,4 +2,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 resources:
   - et-sb-preview.enc.yaml
+  - et-postgres.enc.yaml
```
Comment (hsjhita) [inline apps/et/preview/sops-secrets/kustomization.yaml:5]: need to add this file in sops-secrets

File: apps/et/preview/aso/et-postgres.yaml (hunk 1 above)
Comment (timja) [reply apps/et/preview/aso/et-postgres.yaml:12]:
```suggestion
  version: "15"
```

File: apps/et/preview/aso/et-postgres.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+apiVersion: dbforpostgresql.azure.com/v1api20210601
+kind: FlexibleServer
+metadata:
+  name: ${NAMESPACE}-flexibleserver-postgres-${ENVIRONMENT}
+  namespace: ${NAMESPACE}
+  annotations:
+    serviceoperator.azure.com/reconcile-policy: detach-on-delete
+spec:
```
Comment (timja) [inline apps/et/preview/aso/et-postgres.yaml:8]:
```suggestion
spec:
  tags:
    builtFrom: https://github.com/hmcts/cnp-flux-config
    businessArea: CFT
    environment: development
    autoShutdown: "true"
    application: employment-tribunals
```

**Review by hsjhita — APPROVED** [review]
Tim suggested Postgres version 15 but LGTM

**Review by hsjhita — APPROVED** [review]

**Comment by timja** [comment]
> Tim suggested Postgres version 15 but LGTM

15 isn't supported in ASO yet:
https://azure.github.io/azure-service-operator/reference/dbforpostgresql/v1api20210601/#dbforpostgresql.azure.com/v1api20210601.ServerVersion

File: apps/et/preview/aso/et-postgres.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+apiVersion: dbforpostgresql.azure.com/v1api20210601
+kind: FlexibleServer
+metadata:
+  name: ${NAMESPACE}-${ENVIRONMENT}
+  namespace: ${NAMESPACE}
+  annotations:
+    serviceoperator.azure.com/reconcile-policy: detach-on-delete
```
Comment (timja) [inline apps/et/preview/aso/et-postgres.yaml:7]: no need to duplicate this
```suggestion
```

**Review by timja — APPROVED** [review]

**Review by hsjhita — APPROVED** [review]

**Pull request merged**